### ISSUE TITLE
Documentation on default ldap port incorrect

### ISSUE THREAD

**Issue by phemmer** [grafana/grafana#6404]
* **I'm submitting a ...**
- [x] Bug report
- [ ] Feature request
- [ ] Question / Support request: **Please do not** open a github issue. [Support Options](http://grafana.org/support/)

Please include this information:
- What Grafana version are you using?
grafana-3.1.1-1470047149

- What datasource are you using?
n/a

- What OS are you running grafana on?
linux

- What did you do?
Configured LDAP with `port` undefined.

- What was the expected result?
Working ldap

- What happened instead?

    t=2016-10-27T13:23:23-0400 lvl=eror msg="Error while trying to authenticate user" logger=context userId=0 orgId=0 uname= error="LDAP Result Code 200 \"\": dial tcp 10.0.2.12:0: getsockopt: connection refused"

- Issue description:
The documentation at http://docs.grafana.org/installation/ldap/ claims that the default port is 389. However as seen from the above error, grafana tries to use port 0.

**Comment by torkelo** [comment]
no, port 389 is the default specified in ldap.toml

**Comment by torkelo** [comment]
and that is the port being using when I am testing it now, maybe the connection error is misleading, not sure where it comes from (golang socket code I guess)

**Comment by phemmer** [comment]
I beg to differ. Here's a packet capture:

```
16:31:42.536590 IP 10.0.33.104.37024 > 10.0.2.12.0: Flags [S], seq 3942089766, win 29200, options [mss 1460,sackOK,TS val 2483135178 ecr 0,nop,wscale 7], length 0
16:31:42.536771 IP 10.0.2.12.0 > 10.0.33.104.37024: Flags [R.], seq 0, ack 3942089767, win 0, length 0
```

Notice the destination, `10.0.2.12.0`. This means it tried to connect to `10.0.2.12` port `0`.

**Comment by torkelo** [comment]
Very strange, your ldap.toml file looks like?

**Comment by phemmer** [comment]
```
[[servers]]
host = "myldap.mycompany.com"
bind_dn = "CN=foo,dc=mycompany,dc=com"
bind_password = 'xyzzy'
search_base_dns = ["ou=HQ,dc=mycompany,dc=com"]
#port = 389

search_filter = "(sAMAccountName=%s)"

[servers.attributes]
name = "cn"
surname = "sn"
username = "sAMAccountName"
member_of = "memberOf"
email = "mail"


[[servers.group_mappings]]
group_dn = "CN=xyz"
org_role = "Admin"

[[servers.group_mappings]]
group_dn = "*"
org_role = "Viewer"
```

**Comment by torkelo** [comment]
You seem to have commented out the port option

**Comment by phemmer** [comment]
That's the point of this issue. 

> What did you do?
Configured LDAP with port undefined.

&nbsp;

> Issue description:
The documentation at http://docs.grafana.org/installation/ldap/ claims that the default port is 389. However as seen from the above error, grafana tries to use port 0.

**Comment by torkelo** [comment]
The documentation is a bit miss leading, what it means is that 389 is usually the default ldap port used by ldap servers, and 636 is usually used by ldap servers where ssl is enabled.

**Comment by phemmer** [comment]
So, can we get it fixed?

**Comment by MattSANU** [comment]
This is still an issue in Grafana 4.5.1. The default port is still 0, and the documentation still falsely claims that the default is not zero.

Has this been fixed in later versions of Grafana? Version 4.6.1 is the latest at time of writing. If not, is there a reason this issue has been closed?

I perceive multiple problems here:
- Inline documentation in ldap.toml is a good place to document the operation of Grafana. If the 'port' option has a default value, then that default value should be documented in that file. Currently the default is '0', but that fact is not documented in that file.
- The grafana documentation is not the correct place to describe the ports that LDAP servers commonly use - the RFCs are the correct place for that. RFC4510 et seq (especially 4511) seem sufficient; there is no need to repeat them in comments in a Grafana configuration file. The current comment seems to be an attempt to help someone deploying Grafana guess what port(s) their LDAP server(s) are running on. Encouraging guesswork isn't helpful.
- 0 is not a good default value for the 'port' option. This is not a common port on which to serve LDAP. Also, port 0 is reserved by RFC 170, so software such as Grafana should not be using it. Either there should be no default value for the 'port' option, in which case the 'port' option should be required, or if it continues to have a default value, then that default value should be obtained using a call to netdb.GetServByName, with name "ldap" if use_ssl is falsy, or "ldaps" if use_ssl is truthy.
